### PR TITLE
feat: busyTimeout

### PR DESCRIPTION
**Pull request by ronag**
Refs: https://github.com/mcollina/undici/issues/310

### CODE REVIEW

**Comment by ronag** [comment]
Added nowCache. It's slower than master...

```js
http - keepalive x 5,630 ops/sec ±2.10% (276 runs sampled)
undici - pipeline x 8,961 ops/sec ±1.57% (270 runs sampled)
undici - request x 12,344 ops/sec ±0.62% (279 runs sampled)
undici - stream x 13,533 ops/sec ±0.80% (276 runs sampled)
undici - dispatch x 14,498 ops/sec ±0.43% (279 runs sampled)
undici - noop x 22,865 ops/sec ±1.18% (258 runs sampled)

root@dc8f9156a539:/undici# git checkout master
Switched to branch 'master'
Your branch is up-to-date with 'origin/master'.
root@dc8f9156a539:/undici# node benchmarks/index.js 

http - keepalive x 5,707 ops/sec ±1.79% (270 runs sampled)
undici - pipeline x 9,333 ops/sec ±1.94% (278 runs sampled)
undici - request x 12,606 ops/sec ±0.72% (279 runs sampled)
undici - stream x 13,805 ops/sec ±0.61% (280 runs sampled)
undici - dispatch x 15,050 ops/sec ±0.53% (279 runs sampled)
undici - noop x 23,954 ops/sec ±1.21% (257 runs sampled)

```

File: lib/util.js [hunk 1]
```diff
@@ -88,6 +88,22 @@ function isBuffer (buffer) {
...
+let nowCache
+
+function nowDate () {
+  if (!nowCache) cache()
+  return nowCache
+}
+
+function cache () {
+  nowCache = Date.now()
+  setTimeout(resetCache, 1000 - nowCache).unref()
```
Comment (ronag) [inline lib/util.js:100]: Maybe setInterval might be faster?

**Comment by ronag** [comment]
setInterval looks better:

```js
http - keepalive x 5,904 ops/sec ±1.83% (274 runs sampled)
undici - pipeline x 8,808 ops/sec ±1.99% (269 runs sampled)
undici - request x 12,753 ops/sec ±0.60% (280 runs sampled)
undici - stream x 13,723 ops/sec ±0.69% (281 runs sampled)
undici - dispatch x 15,047 ops/sec ±0.52% (278 runs sampled)
```

**Comment by ronag** [comment]
or maybe it's within margin of error... unclear

**Comment by mcollina** [comment]
I think the `setTimeout` approach is better (if we need it at all). I prefer not to wake up a process every second just for requiring this module.

**Comment by ronag** [comment]
blocked by https://github.com/mcollina/undici/issues/310#issuecomment-675487008

**Pull request closed without merging**